VerilogCodingStyle.md

```diff
@@ -965,12 +965,14 @@ facilitate the creation of IP that may be re-used across many projects.
 
 The preferred naming convention for all immutable constants is to use `ALL_CAPS`, but there are times when the use of `UpperCamelCase` might be considered more natural.
 
+The naming convention for parameter and localparam allows for the use of `ALL_CAPS` or `UpperCamelCase`. The allowance for `ALL_CAPS` is to accommodate this ubiquitous style found in the ASIC industry, and by all third party IP vendors.
```

Comment:
I have no problem with allowances for `ALL_CAPS`, but I think our code base
is preferring `UpperCamelCase`, so that should be our leader. Plus you've
undermined our past decisions a bit with the wording. This sentence plus
the table below imply that `ALL_CAPS` is preferred, which isn't what we've
agreed upon. How about this as a suggestion:

The naming convention for `parameter` and `localparam` allows for the use of `ALL_CAPS` or `UpperCamelCase`.
`UpperCamelCase` is the preferred style for mutable parameters, but there is allowance for `ALL_CAPS` to accommodate the common style found in other code bases.

Reply:
I totally agree with @sjgitty here. `UpperCamelCase` should be clearly marked as preferred style.

Reply:
As for allowing the use of `ALL_CAPS` for (local) parameters: I am against it. Once people start to use both, it won't be any more distinguishable if something is constant or tunable during code review. And this was the whole point of the original discussion. But this is my personal opinion. I would like to hear other peoples voices.